File: vriatui-admin-ui/src/app/models/incident.ts
import { Profile } from "./profile";
import { Address } from "./address";

export class Incident {
    id: string;
    location: string; // GeoJsonPoint
    address: Address;
    problem : string;
    happenAt: Date; //LocalDateTime
    patientInfo : Profile;

    constructor() {
    }
}

File: src/main/java/com/testname/vriatui/model/Incident.java
package com.testname.vriatui.model;

import lombok.Data;
import org.springframework.data.annotation.Id;

import java.time.LocalDateTime;

@Data
public class Incident {
    @Id
    private String id;
    private Address address;
    private String problem;
    private LocalDateTime happenAt;
    private PatientInfo patient;

    public Incident() {
    }

    public Incident(IncidentRequest incidentRequest, Profile profile) {
        this.address = incidentRequest.isHappenAtHome()
                ? profile.getAddress()
                : new Address(incidentRequest.getLocation());
        this.happenAt = LocalDateTime.now();
        this.problem = incidentRequest.getProblem();
        this.patient = new PatientInfo(profile);
    }
}
